elastic/beats
# Add a fully-featured dashboard for Enterprise Search

## What does this PR do?

This PR adds a fully-featured Kibana dashboard to be used with the new Enterprise Search module for metricbeat.
See the Screenshots section below for an example screenshot.

## Why is it important?

Enterprise Search users need to be able to visualize the metrics produced by the new Enterprise Search metricbeat module. Since the Stack Monitoring integration is unlikely to land in 7.16 (https://github.com/elastic/kibana/pull/114303), we want to ship the Kibana dashboard version of the same visualizations to make it easier to our users to consume new metrics in an efficient way.

## Checklist


## Author's Checklist

- [x] Tested the new dashboard by importing it via the metricbeat setup command

## How to test this PR locally

1. Go to `x-pack/metricbeat`
2. Build metricbeat via `mage build`
3. Import the dashboards:

```
./metricbeat setup --dashboards -E setup.dashboards.enabled=true -E setup.dashboards.always_kibana=true -E setup.kibana.host=http://localhost:5601/xtu -E setup.kibana.username=elastic -E setup.kibana.password=changeme -E setup.dashboards.directory=build/kibana -e -v
```

4. Run metricbeat with the enterprise search module against an Enterprise Search instance
5. Open Kibana and see some glorious graphs!

## Screenshots

![localhost_5601_xtu_app_dashboards](https://user-images.githubusercontent.com/3467/137214721-86427655-2648-4c77-b1ad-404b9f0375ee.png)
